### PR TITLE
MM-47980, MM-47981 Improve team limits

### PR DESCRIPTION
**Pull request by crspeller**
## Summary
Improvements to the team limitation functionality. See tickets. 

## Ticket Link
https://mattermost.atlassian.net/browse/MM-47980
https://mattermost.atlassian.net/browse/MM-47981

### CODE REVIEW

File: server/sqlstore/playbook_run.go [hunk 1]
```diff
@@ -983,18 +983,32 @@ func (s *playbookRunStore) buildPermissionsExpr(info app.RequesterInfo) sq.Sqliz
 		))`, info.UserID, info.UserID)
 }
 
-func buildTeamLimitExpr(userID, teamID, tableName string) sq.Sqlizer {
-	if teamID != "" {
-		return sq.Eq{fmt.Sprintf("%s.TeamID", tableName): teamID}
-	}
-
-	return sq.Expr(fmt.Sprintf(`
+func buildTeamLimitExpr(info app.RequesterInfo, teamID, tableName string) sq.Sqlizer {
```
Comment (iomodo) [inline server/sqlstore/playbook_run.go:986]: We need only userID from `RequesterInfo` right? Why change?

**Review by iomodo — APPROVED** [review]
LGTM! Minor comment below

File: server/sqlstore/playbook_run.go (hunk 1 above)
Comment (crspeller) [reply server/sqlstore/playbook_run.go:986]: We also use `info.IsAdmin`

File: server/sqlstore/playbook_test.go [hunk 2]
```diff
@@ -815,10 +815,10 @@ func TestGetPlaybooksForTeam(t *testing.T) {
 				PerPage: 1000,
```
Comment (lieut-data) [inline server/sqlstore/playbook_test.go:815]: There are a lot of commented out tests above -- wonder if this is an opportunity to holistically update everything?

**Review by lieut-data — APPROVED** [review]

**Review by Phrynobatrachus — APPROVED** [review]

**Pull request merged**